aws/sns/resources.py

```diff
@@ -21,20 +22,49 @@ def sns_topics():
     )
 
 
-def sns_subscription_attributes():
+def sns_parent_topic_exists(subscription):
     "https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sns.html#subscription"
-    return [
+    topicArn = ":".join(
+        re.split(":", subscription)[0:6]
+    )  # Naive method of divining topic from subscription
+    try:
         botocore_client.get(
             service_name="sns",
-            method_name="get_subscription_attributes",
+            method_name="get_topic_attributes",
             call_args=[],
-            call_kwargs={"SubscriptionArn": subscription["SubscriptionArn"]},
-            profiles=[subscription["__pytest_meta"]["profile"]],
-            regions=[subscription["__pytest_meta"]["region"]],
+            call_kwargs={"TopicArn": topicArn},
+        )
```

Comment:
We have some kind of caching of API results, right? So maybe it would be nicer to call `list_topics` here, and check whether the topic is in the list. If the result of `list_topics` is indeed cached, we'd only make a single API call for all subscriptions, instead of one for each subscription.

Reply:
I had thought about doing that originally, and not being pythonically inclined, I couldn't think of how to go about doing it.  However, as you suggest, it really is a better solution and I _think_ I can mange it.

Reply:
If the API call is implicitly cached, you don't need to change much – just call `list_topics` here every time this is called, and the list will be retrived from the cache. However, I'm not sure how that weird cache works, so you'd need to figure out whether it's actually cached.